src/main/java/roboy/util/TelegramCommunicationHandler.java

```diff
@@ -0,0 +1,339 @@
+        if(telegramTimeouts.isEmpty()){
+            Timeout t  = new Timeout(millisecond * INPUT_TIME_LIMIT);
+            t.setUnique(chatID);
+
+            telegramTimeouts.add(t);
+        }
+
+        boolean hasStarted = false;
+        for(Timeout t: telegramTimeouts){
+            if(t.getUnique().equals(chatID)){
+                hasStarted = true;
+                t.start(this);
+            }
+        }
+
+        if(!hasStarted){
+            // there is no timeout for given chatID so start one
+            Timeout t  = new Timeout(millisecond * INPUT_TIME_LIMIT);
+            t.setUnique(chatID);
+            t.start(this);
+
+            telegramTimeouts.add(t);
+        }
+    }
+
+    /**
+     * Receives the updates from telegram's api and called by it.
+     * @param update consist of an update of a chat.
+     */
+    @Override
+    public void onUpdateReceived(Update update) {
+        if(update.hasMessage()){
+            Message message = update.getMessage();
+            User user = message.getFrom();
+
+            if(message.hasText()){
+                String chatID = message.getChatId().toString();
+                String text = message.getText();
+                if(text.startsWith("/")){
+                    //inline command
+                }else{
+                    try {
+                        //get message, add it to containers
+                        //pairs.add(new Pair<>(chatID, text));
+
+                        HashMap<String, String> pair = new HashMap<String, String>();
+                        pair.put(chatID, text);
+                        pairs.add(pair);
+
+                        //wait for certain seconds, start the timer
+                        handleTimeout(chatID);
+
+                    } catch (Exception e) {
+                        Log.error(this, "Message receiving has been interrupted.");
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }else{
+            //The update does not have messages
+            Log.error(this,"update: "+update.toString());
+        }
+    }
+
+    /**
+     * Called when specified time passed w.r.t. unique chatID.
+     * Concatenate all the messages that is not processed.
+     * Calls the InputDevice for telegram
+     * @param chatID unique identifier for a chat.
+     */
+    @Override
+    public void onTimeout(String chatID) {
+
+        // timeout method for getting all the messages
+        //telegramTimeout.stop();
+        for(Timeout t: telegramTimeouts){
+            if(t.getUnique().equals(chatID)){
+                t.stop();
+            }
+        }
+
+        List<HashMap<String, String>> removedObjects = new ArrayList<>();
+
+        // get the all messages
+        HashMap<String, String> result = null;
+        for(HashMap<String, String> p: pairs){
+            //Map a = new HashMap<String, String>();
+            String chatIdFromPair = p.keySet().toArray()[0].toString();
```

Comment:
HashMaps as pairs are not nice 
`p.keySet().toArray()[0].toString()` o_O
Create a custom Pair class for this